tests/tests/tier0/bluechi-is-online-system-monitor/test_bluechi_is_online_system_monitor.py

```diff
@@ -0,0 +1,164 @@
+#
+# Copyright Contributors to the Eclipse BlueChi project
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+import logging
+import threading
+import time
+from typing import Dict
+
+from bluechi_test.config import BluechiAgentConfig, BluechiControllerConfig
+from bluechi_test.machine import BluechiAgentMachine, BluechiControllerMachine
+from bluechi_test.test import BluechiTest
+from bluechi_test.util import Timeout, get_test_env_value_int
+
+LOGGER = logging.getLogger(__name__)
+
+AGENT_ONE = "agent-one"
+AGENT_TWO = "agent-two"
+SLEEP_DURATION = get_test_env_value_int("SLEEP_DURATION", 2)
+
+
+class MonitorResult:
+    def __init__(self):
+        self.result = None
+        self.output = ""
+
+
+def monitor_command(ctrl: BluechiControllerMachine, monitor_result: MonitorResult):
+    monitor_result.result, monitor_result.output = (
+        ctrl.bluechi_is_online.monitor_system()
+    )
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    # Test 1: Both agents and controller are online
+    LOGGER.debug("Testing with both agents and controller online.")
+    monitor_result_test_one = MonitorResult()
+    monitor_thread = threading.Thread(
+        target=monitor_command, args=(ctrl, monitor_result_test_one)
+    )
+    monitor_thread.start()
+    try:
+        with Timeout(SLEEP_DURATION, "Timeout while monitoring the system."):
+            monitor_thread.join()
+    except TimeoutError:
+        LOGGER.debug(
+            "Timeout reached while monitoring system. Attempting to terminate."
+        )
+
+    assert (
+        monitor_result_test_one.result is None
+    ), "Monitor command should not produce output when system is running."
+
+    # Test 2: Agent-one offline
+    LOGGER.debug("Stopping agent-one.")
+    nodes[AGENT_ONE].systemctl.stop_unit("bluechi-agent")
+
+    assert nodes[AGENT_ONE].wait_for_unit_state_to_be("bluechi-agent", "inactive")
+
+    monitor_result_test_two = MonitorResult()
+    monitor_thread = threading.Thread(
+        target=monitor_command, args=(ctrl, monitor_result_test_two)
+    )
+    monitor_thread.start()
+    time.sleep(SLEEP_DURATION)
+    monitor_thread.join()
```

Comment:
If one agent is down, the `bluechi-is-online system --monitor` should exit immediately. So I think we can remove that `time.sleep` here:
```suggestion
    monitor_thread.start()
    monitor_thread.join()
```

Same in test cases 3, 4, 5.

Reply:
Done